### ISSUE TITLE
Old array syntax for match appears deprecated, but still documented.

### ISSUE THREAD

**Issue by jarpy** [logstash-plugins/logstash-filter-grok#44]
The [docs](https://www.elastic.co/guide/en/logstash/current/plugins-filters-grok.html#plugins-filters-grok-match) suggest that the `match` configuration option can still take an array, but the current code doesn't seem to like it.

Repro script:
```bash
echo 'banana 7' | /opt/logstash/bin/logstash -e '
input {
  stdin {}
}

filter {
  grok {
    match => [
      "message",
      "%{WORD:word}",
      "%{POSINT:number}"
    ]
  }
}

output {
  stdout {
    codec => rubydebug
  }
}
'
```
Output:
```
Invalid setting for grok filter plugin:

  filter {
    grok {
      # This setting must be a hash
      # This field must contain an even number of items, got 3
      match => ["message", "%{WORD:word}", "%{POSINT:number}"]
      ...
    }
  } {:level=>:error}
```

**Comment by jarpy** [comment]
#45

**Comment by jarpy** [comment]
Fixed in #46
